pkg/strc/middleware.go

```diff
@@ -93,6 +114,14 @@ func NewMiddlewareWithConfig(logger *slog.Logger, config MiddlewareConfig) func(
 			// dump response body
 			bw := newBodyWriter(w, ResponseBodyMaxSize, config.WithResponseBody)
 
+			// apply filters early
+			for _, filter := range config.Filters {
+				if !filter(bw, r) {
+					next.ServeHTTP(bw, r)
+					return
+				}
+			}
+
```

Comment:
Moved filters to exit earlier, so paths like `/metrics` do not have any trace/span processing at all, they do not need to.